### PR TITLE
Reverse geocoder

### LINKED ISSUE #117: Reverse geocoder
The reverse geocoder takes as an input the `lat,lon` for a observation (eg location of university campus) and determines its geography using the two geographies that we are focusing on in the project (NUTS2 and LEPS)

- [x] Write and test prototype of reverse geocoder in notebook
- [ ] Create package

### PR DESCRIPTION
**Pull request by Juan-Mateos**
Closes #117 

This is an implementation of a reverse geocoder that we use to extract the NUTS2 region of a location for different years. We have used it to create a `uni_geos.json` object with nuts codes for universities according to different NUTS definitions.

### CODE REVIEW

File: ds/beis_indicators/geo/reverse_geocoder.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    
+    print('Joining...')
+    
+    #Spatial join: looks for points inside the polygons
+    joined = gp.sjoin(place_holder,shape,op='within')
+    
+    #Return the joined df
+    return(joined)
+
+#get_shape('nuts2_2010',f'{project_dir}/data/shapefiles/')
```
Comment (mindrones) [inline ds/beis_indicators/geo/reverse_geocoder.py]: leftover?

File: ds/beis_indicators/geo/university_reverse_geocode.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+with open(f'{project_dir}/data/aux/shapefile_urls.json','r') as infile:
+    shape_lookup = json.load(infile)
+
+#Get all the shapefiles
+for name in shape_lookup.keys():
+    get_shape(name,path=f'{project_dir}/data/raw/shapefiles/')
+
+#University metadata
+uni_meta = pd.read_csv(
+                       'http://learning-provider.data.ac.uk/data/learning-providers-plus.csv')
```
Comment (mindrones) [inline ds/beis_indicators/geo/university_reverse_geocode.py]: I'd remove these blanks

**Review by mindrones — APPROVED** [review]
Seems good to merge to me but I haven't tested it locally.

A couple of doubts:
- is `university_reverse_geocode.py` supposed to be partt of this PR?
- are you suppose to do the `Create package` subtask?

**Comment by Juan-Mateos** [comment]
I included `university_reverse_geocode.py` as a test for the `reverse_geocoder.py`. Since I am using the rev geocoded university files in multiple issues it felt like a good idea to do & pull here.

**Pull request merged**